File: server/src/main/java/com/example/server/note/domain/Note.java
package com.example.server.note.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@Getter
@Entity
public class Note extends Date{
    @Id
    private String slug;

    @Column(nullable = false)
    private String name;

    @Column
    private String description;
}

File: server/server/src/main/kotlin/work/notespace/server/note/entity/Note.kt
package work.notespace.server.note.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import work.notespace.server.note.dto.NoteDto

@Entity
class Note(
    @Id
    @Column
    val slug: String,

    @Column(nullable = false)
    var name: String,

    @Column
    var description: String,
) : Date() {
    fun toDto(): NoteDto {
        return NoteDto(
            slug = slug,
            name = name,
            description = description,
        )
    }
}
